stl/inc/execution

```diff
@@ -5050,6 +5051,170 @@ _FwdIt2 adjacent_difference(_ExPo&&, const _FwdIt1 _First, const _FwdIt1 _Last,
                              _Get_unwrapped_n(_Dest, _Idl_distance<_FwdIt1>(_UFirst, _ULast)), _Pass_fn(_Diff_op)));
     return _Dest;
 }
+
+_EXPORT_STD template <class _ExPo, class _NoThrowFwdIt, _Enable_if_execution_policy_t<_ExPo> /* = 0 */>
+void destroy(_ExPo&& _Exec, const _NoThrowFwdIt _First, const _NoThrowFwdIt _Last) noexcept /* terminates */ {
+    // destroy all elements in [_First, _Last)
+    _REQUIRE_PARELLEL_LVALUE_ITERATOR(_NoThrowFwdIt);
+    using _Ty = remove_reference_t<_Iter_ref_t<_NoThrowFwdIt>>;
+    _STD _Adl_verify_range(_First, _Last);
+    if constexpr (!is_trivially_destructible_v<_Ty>) {
+        _STD for_each(_STD forward<_ExPo>(_Exec), _First, _Last, [](_Ty& _Obj) { _STD _Destroy_in_place(_Obj); });
+    }
+}
+
+_EXPORT_STD template <class _ExPo, class _NoThrowFwdIt, class _Diff, _Enable_if_execution_policy_t<_ExPo> /* = 0 */>
+_NoThrowFwdIt destroy_n(_ExPo&& _Exec, _NoThrowFwdIt _First, const _Diff _Count_raw) noexcept /* terminates */ {
+    // destroy all elements in [_First, _First + _Count)
+    _REQUIRE_PARELLEL_LVALUE_ITERATOR(_NoThrowFwdIt);
+    using _Ty                      = remove_reference_t<_Iter_ref_t<_NoThrowFwdIt>>;
+    _Algorithm_int_t<_Diff> _Count = _Count_raw;
+    if (_Count <= 0) {
+        return _First;
+    }
+
+    if constexpr (is_trivially_destructible_v<_Ty>) {
+        _STD advance(_First, _Count);
+        return _First;
+    } else {
+        return _STD for_each_n(
+            _STD forward<_ExPo>(_Exec), _First, _Count, [](_Ty& _Obj) { _STD _Destroy_in_place(_Obj); });
+    }
+}
+
+_EXPORT_STD template <class _ExPo, class _NoThrowFwdIt, _Enable_if_execution_policy_t<_ExPo> /* = 0 */>
+void uninitialized_default_construct(_ExPo&& _Exec, const _NoThrowFwdIt _First, const _NoThrowFwdIt _Last) noexcept
+/* terminates */ {
+    // default-initialize all elements in [_First, _Last)
+    _REQUIRE_PARELLEL_LVALUE_ITERATOR(_NoThrowFwdIt);
+    using _Ty = _Iter_value_t<_NoThrowFwdIt>;
+    _STD _Adl_verify_range(_First, _Last);
+    if constexpr (!is_trivially_default_constructible_v<_Ty>) {
+        _STD for_each(_STD forward<_ExPo>(_Exec), _First, _Last,
+            [](auto& _Obj) { _STD _Default_construct_in_place(reinterpret_cast<_Ty&>(_Obj)); });

```

Comment:
This was changed due to https://github.com/microsoft/STL/pull/3145#discussion_r1047875148.

Reply:
Ok, thanks. I'll see if I can get an example from Casey, good to keep as-is.

Reply:
Yeah, this looks like a misreading of the standard to me; we should not be `reinterpret_cast`ing here.

Reply:
After much discussion in Discord, in conjunction with pre-knowledge that LEWG has voted to get rid of _`voidify`_ in C++23, I'm convinced that I was wrong. I'll push a reversion since I feel bad asking a contributor to do work to undo other work that I asked for ;)